torch/csrc/jit/script/sugared_value.h

```diff
@@ -250,16 +255,20 @@ struct MethodValue : public SugaredValue {
       at::ArrayRef<NamedValue> inputs,
       at::ArrayRef<NamedValue> attributes,
       size_t n_binders) override {
-    std::vector<NamedValue> inputsWithSelf;
-    inputsWithSelf.emplace_back(loc, self_.value(*f.graph()));
+    std::vector<NamedValue> inputsWithSelf = {self_};
     inputsWithSelf.insert(inputsWithSelf.end(), inputs.begin(), inputs.end());
-    return FunctionValue(method_).call(
-        loc, f, inputsWithSelf, attributes, n_binders);
+    auto method = self_->type()->expect<ClassType>()->getMethod(method_name_);
+    TORCH_INTERNAL_ASSERT(method);
+    method->ensure_defined();
```

Comment:
It seems we are frequently calling `ensure_defined()` before doing something with a method—I think lazy compilation should be an internal detail of `Method` and callers shouldn't need to know about it. Don't have to change it here though.

Reply:
I agree with this, but I don't know of a great place to put it yet. It is possible that we can choose which methods on Function should just call it but I haven't figured out if that works.